### PR TITLE
Adding general try-except (SCP-4279)

### PR DESCRIPTION
**Pull request by devonbush**
This adds a general try except around all ingest code, with the hope of getting better sentry visibility into what's going on with lower level errors. 

TO TEST:
1. confirm a normal ingest runs successfully.
2. simulate a low-level DB connection initializing error by  inserting `raise NameError('Test error')` as the first line of `get_mongo_db` on line 163 of `ingest_pipeline.py`
3. add a breakpoint/log statement in `log_exception` of `monitor.py`
4. rerun an ingest, confirm breakpoint/log is triggered

### CODE REVIEW

File: ingest/ingest_pipeline.py [hunk 1]
```diff
@@ -576,21 +563,27 @@ def main() -> None:
...
+            arguments["user_metrics_uuid"],
+        )
+        ingest = IngestPipeline(**arguments)
+        status, status_cell_metadata = run_ingest(ingest, arguments, parsed_args)
+        # Print metrics properties
+        metrics_dump = config.get_metric_properties().get_properties()
+        for key in metrics_dump.keys():
+            print(f'{key}: {metrics_dump[key]}')
+    except Exception as e:
+        # ToDo ingest.props["errorType"] = "subsample:"
```
Comment (eweitz) [inline ingest/ingest_pipeline.py]: Please note ticket for TODO in this code comment.

File: schema/README.md [hunk 2]
```diff
@@ -7,19 +7,22 @@
...
   ```
-  
-* Copy the new convention JSON and TSV files to the * &lt;project&gt;_convention directory  
-  
+
+* Copy the new convention JSON and TSV files to the * &lt;project&gt;_convention directory
+
+* Update the corresponding files, if appropriate, in single_cell_portal_core/lib/assets/metadata_schemas
+
+* If required columns have changed, update REQUIRED_CONVENTION_COLUMNS in single_cell_portal_core/app/javascript/lib/validation/single_cell_portal_core/validate-file-content.js
```
Comment (eweitz) [inline schema/README.md:25]: At a glance, these lines look unrelated to this issue and aren't mere syntax changes, which makes me skittish.

File: ingest/ingest_pipeline.py [hunk 3]
```diff
@@ -576,24 +576,29 @@ def main() -> None:
...
+        status, status_cell_metadata = run_ingest(ingest, arguments, parsed_args)
+        # Print metrics properties
+        metrics_dump = config.get_metric_properties().get_properties()
+        for key in metrics_dump.keys():
+            print(f'{key}: {metrics_dump[key]}')
+
+        # Log Mixpanel events
+        MetricsService.log(config.get_parent_event_name(), config.get_metric_properties())
+    except Exception as e:
+        log_exception(IngestPipeline.dev_logger, IngestPipeline.user_logger, e)
```
Comment (eweitz) [inline ingest/ingest_pipeline.py:600]: Am I right that this logs only to files, and not to Sentry nor Mixpanel?

**Review by eweitz — APPROVED** [review]

File: ingest/ingest_pipeline.py (hunk 3 above)
Comment (devonbush) [reply ingest/ingest_pipeline.py:600]: this logs to Sentry and to files, but not to Mixpanel

File: ingest/ingest_pipeline.py (hunk 1 above)
Comment (devonbush) [reply ingest/ingest_pipeline.py]: comment was pre-existing to this PR

**Review by jlchang — APPROVED** [review]

**Review by ehanna4 — APPROVED** [review]
Code change makes sense to me 👍

**Comment by eweitz** [comment]
For reference, [this Slack thread](https://broadinstitute.slack.com/archives/CBEHTH601/p1650048785594079) has more discussion, logs, charts, etc. related to the following error that prompted this PR:

```
pymongo.errors.ServerSelectionTimeoutError: 10.128.0.92:27017: timed out
```
